plugins/inputs/jolokia2/gatherer.go

```diff
@@ -88,8 +88,8 @@ func (g *Gatherer) generatePoints(metric Metric, responses []ReadResponse) ([]po
 		case 404:
 			continue
 		default:
-			errors = append(errors, fmt.Errorf("Unexpected status in response from target %s: %d",
-				response.RequestTarget, response.Status))
+			errors = append(errors, fmt.Errorf("Unexpected status in response from target %s (%q): %d",
+				response.RequestTarget, response.RequestMbean, response.Status))
 			continue
 		}
 
```

Comment:
How to use this code in my server where telegraf is installed. Please suggest.

Reply:
Download a nightly build and test it out. https://github.com/influxdata/telegraf#nightly-builds

Reply:
I am getting below error with nightly build log. I don't understand what's wrong in this configuration during collection of undertow subsystem. Except **undertow** all other subsystems able to collect data from target server by jolokia plugin. Please help!!

 ### 2020-08-13T10:43:10Z E! [inputs.jolokia2_agent] Error in plugin: Unexpected status in response from target  ("jboss.as:http-listener=*,server=*,subsystem=undertow"): 400

Reply:
a 400 error means "bad request". This means the server couldn't understand what you're requesting, and I would suspect the mbean request. Maybe that the server doesn't understand `*`, or that it's not escaped correctly by the plugin.

I don't know why you're not seeing any output. You could try viewing it before it goes to the InfluxDB server like this:
```
[[outputs.file]]
  files = ["stdout"]
```

Reply:
It couldn't helped me as well. I didn't find any information regarding undertow subsystems for "connectors_http" in output file. Even also no information regarding such in telegraf log file. Please enhance the log by adding few contexts, why it is not collecting any data in the case of no error from the plugin site. Please Help!!!!

For the meanwhile I am looking for * symbol issue from server end.